PostBuild/CopyUpgrades.cs

```diff
@@ -231,7 +231,7 @@ private static void CollectMethodVersioning(BsonDocument content)
             {
                 foreach (PreviousVersionAttribute attribute in method.GetCustomAttributes<PreviousVersionAttribute>())
                 {
-                    if (attribute != null && attribute.FromVersion == version)
+                    if (attribute != null && attribute.FromVersion == version && toNewContent.All(x => x.Name != attribute.PreviousVersionAsText))
```

Comment:
This I am not sure of. I take that this addition is basically saying "do not add if it has already exists"? 
Firstly, this feels like out of scope of what this PR is meant to do unless I am missing something?
More important though is IMO, this should _probably_ be the other way I round I would have thought, as in if it already exists, it should be replaced by the new versioning coming in.
An example of this could for example be a method that is changed more than once in a milestone. Then ofc this post build needs to collect the latest and not keep a previous already existing versioning key there.

I _think_ this would not matter for installers (@FraserGreenroyd to correct me if I am correct or not here) as they start from a clean slate anyway and rebuild everything while being produced, but could lead to annoyance and confusion for us devs.

TL;DR: Would replace old content rather than keep the old and ignore new if there is this kind of clash.

@adecler and @FraserGreenroyd good to get you to chip in here as well.

Reply:
I think you are right, I did not think it through well enough for one main reason: this issue had never occurred before. If it did, the postbuild would simply have crashed on the attempt to add the same element twice. This is what started happening when I was working on this PR, caused by Revit_Core_Engine being built per each version of Revit, resulting in multiple same named classes/methods that need to be versioned.

So I think the right approach would be to overwrite the previous version, as you suggest.

Reply:
Sure thing @FraserGreenroyd, happy to put as many comments as required